libibverbs/device.c

```diff
@@ -262,6 +267,8 @@ struct ibv_context *__ibv_open_device(struct ibv_device *device)
 	context->cmd_fd = cmd_fd;
 	pthread_mutex_init(&context->mutex, NULL);
 
+	ibverbs_device_hold(device);
+
```

Comment:
Open the FD as part of ibv_open_device is too late as it might open the new device and as such openat *won't* fail.  If TS is used by that step, there is no extra need to use open/openat/close.